src/ORM/DataObject.php

```diff
@@ -4348,4 +4349,19 @@ private function getUniqueKeyComponents(): array
     {
         return $this->extend('cacheKeyComponent');
     }
+
```

Comment:
It's not strictly required, and yes we could just put this code in admin UsedOnTable.php, though it's kind of nice having it directly on DataObject.

Having it on DataObject would make it much more 'findable' to other developers and encourage them to use it because they'll get the typehint in their IDE

Would it work if we also put `@internal` on this function for now, along with the `@internal` on the interface?